### PR TITLE
#347 Update Dashboard Design part 1

### PR DESCRIPTION
**Pull request by sandrina-p**
Following #347 , I'm already opening this PR so you guys can follow my process.

I've started by creating the following placeholder components at `/components`:
- `VotingBanner.vue`
- `ProgressOverview.vue`

When finished, this PR will be huge, so should we split it in smaller PRs? I would propose the following:
1. Markup for missing components (`VotingBanner` + `ProgressOverview`) + update `GroupMembers`, `GroupSupportHistory` and `GroupSettings` designs.
2. Update sidebar designs.
3. Update header designs.

#### `VotingBanner.vue` status
- As a first draft, my focus is doing the markup and using _only_ classes from Bulma. So, until then the UI won't be 100% equal to the designs provided on #347.
- Once the markup is all done, the next step would be override Bulma variables/elements to match our design, instead of creating new classes, what do you think? Actually I've already started it in a new file `bulma_on_groupincome/notifications.scss`

#### Bulma notes
- On current installed Bulma version, some helper classes are not available, so I've hardcoded them at `sass/main.scss` so we can use them until we update bulma to the newest version. (After finishing this issue), what do you think?
 
#### Design notes 
- On the VotingBanner, "3 votes received" is a dark gray on the design. Sometimes gray + yellow might be a dangerous combination. To smooth that contrast for us, we can use `opacity` to create a "gray on yellow", so it turns into a  _darker yellow_ instead. We can apply this in any background color and that's what makes "opacity trick" so awesome! There is a cool article about _tips for cheating at design_ and it mention what I've said: [2. Don’t use grey text on colored backgrounds](https://medium.com/refactoring-ui/7-practical-tips-for-cheating-at-design-40c736799886)

With this said, I've added the modifier `is-opacity-n`. That modifier doesn't exist on Bulma at all, but I think it's very useful to better handle contrasts between texts on colored backgrounds.    

----

#### `GroupMembers.vue`
I've updated its layout to match the current designs.

#### `ProgressOverview.vue`
I've started the component. This one is pretty hard to reuse fro Bulma framework, since they don't have any custom elements to handle summaries and progress.
While I'm developing the markup, I'm also thinking on possible improvements that I'll suggest after. My focus now will be to have a layout similar to the current design, so we can iterate over it after.

----

### Thu, 12 April, Update:

So for this first phase:
- Markup for missing components (`VotingBanner` + `ProgressOverview`) 
Update `GroupMembers`, with bulma classes
- `GroupSupportHistory` + `ProgressOverview` with a personalised/themed layout.

I prefer to set this MR as finished and continue the next layout iteration/improvements on a new MR, since this already has a lot of changes. What do you think?

xx

### CODE REVIEW

File: .Gruntfile.babel.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ module.exports = (grunt) => {
         cmd: './node_modules/.bin/mocha --require Gruntfile.js -R spec --bail',
         options: {env: {LOAD_NO_FILE: 'true', ...process.env}}
       },
-      standard: './node_modules/.bin/standard "**/*.{js,vue}"',
+      standard: './node_modules/.bin/standard "**/*.{js,vue}" --fix',
```
Comment (taoeffect) [inline .Gruntfile.babel.js]: `--fix` should only be called intentionally by the developer. If it is placed here it will be run every time a file changes, and that's not good.

Comment (sandrina-p) [reply .Gruntfile.babel.js]: Humm okay, removed, I'll use it only locally

File: frontend/simple/components/VotingBanner.vue [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<template>
+  <div class="notification columns is-multiline is-centered is-warning vb-container">
+    <p class="column is-12">
+      <strong>{{who}}</strong> has proposed changing the <strong>{{what}}</strong> to <strong>{{change}}</strong>.
```
Comment (hubudibu) [inline frontend/simple/components/VotingBanner.vue]: Besides rule changes, there should be 2 more versions of this text for invitation proposals and member removal proposals, something like this:

- "Sam L. proposed inviting Adam H. to the group"
- "Sam L. proposed removing Adam H. from the group"

File: frontend/simple/sass/main.sass [hunk 3]
```diff
@@ -25,9 +25,23 @@ $family-sans-serif: "Helvetica Neue", "Helvetica", "Arial", sans-serif
 
 @import "bulma"
 
+// -----------------------------
+// available on new Bulma version
+// -----------------------------
+
+.is-centered
```
Comment (hubudibu) [inline frontend/simple/sass/main.sass]: unfortunately this will cause problems with our current Bulma version. they do have `is-centered` for columns although it's undocumented, and we're using it for eg. the create group panels - that should be centered as a column, but should not have centered text alignment

File: frontend/simple/components/VotingBanner.vue [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    props: {
+      who: String,
+      what: String,
+      change: String,
+      votesCount: String
+    }
+  }
+</script>
+
+<style lang="scss" scoped>
```
Comment (hubudibu) [inline frontend/simple/components/VotingBanner.vue]: please keep the template - style - script order seen in the other .vue files

File: frontend/simple/sass/bulma_on_groupincome/notifications.scss [hunk 5]
```diff
@@ -0,0 +1,6 @@
+.notification {
+	&.is-warning {
+		background-color: rgba($warning, 0.3);
```
Comment (hubudibu) [inline frontend/simple/sass/bulma_on_groupincome/notifications.scss]: I realized we don't have a styleguide for our CSS yet, any suggestions for that? maybe we could use a linter as well. in the meantime, please use two spaces for indentation

File: frontend/simple/components/GroupMembers.vue [hunk 6]
```diff
@@ -1,36 +1,109 @@
...
+    width: $iconSize;
+    border-radius: 50%;
+
+    .fa.fa-plus {
+      font-size: 2.5rem;
+      height: 2.25rem;
+    }
+
+    // How to I import missins from Bulma to be available here?
+    // Bulma mixins are in SASS but this is SCSS... ._.
```
Comment (hubudibu) [inline frontend/simple/components/GroupMembers.vue]: the syntax difference in itself shouldn't be a problem I think.
we could probably import the bulma mixins file in the components where it's used

File: frontend/simple/components/GroupMembers.vue [hunk 7]
```diff
@@ -1,36 +1,109 @@
...
+    }
   }
 </style>
 <script>
   import {mapGetters} from 'vuex'
   export default {
     name: 'GroupMembers',
+    data () {
+      return {
+        profilesForGroupMock: [{
```
Comment (hubudibu) [inline frontend/simple/components/GroupMembers.vue:92]: 👍 for this mock solution

File: frontend/simple/sass/bulma_on_groupincome/_index.scss [hunk 8]
```diff
@@ -0,0 +1 @@
+@import 'notifications';
```
Comment (hubudibu) [inline frontend/simple/sass/bulma_on_groupincome/_index.scss]: 👍 for keeping the bulma overwrites together

File: frontend/simple/sass/_groupincome.scss [hunk 9]
```diff
@@ -93,3 +93,16 @@
 .is-small {
   font-size: $size-7;
 }
+
+// Have colors variants matching the background
+
+.is-opacity-1 {
```
Comment (hubudibu) [inline frontend/simple/sass/_groupincome.scss]: I know these things don't exist in bulma itself but they are still "theme-related" in a way so maybe they could go together with our bulma overwrites. what do you think @sandrina-p ?

**Review by hubudibu — CHANGES_REQUESTED** [review]
> When finished, this PR will be huge, so should we split it in smaller PRs?

yes, definitely! the separation sounds great.

> the next step would be override Bulma variables/elements to match our design, instead of creating new classes

yes, perfect direction

> On current installed Bulma version, some helper classes are not available, so I've hardcoded them at sass/main.scss so we can use them until we update bulma to the newest version

this is also a good direction, but it has its dangers, see my comment there

great progress so far!  👍 🚀

File: frontend/simple/components/VotingBanner.vue (hunk 2 above)
Comment (sandrina-p) [reply frontend/simple/components/VotingBanner.vue]: Oh I'll update props to handle that part!

File: frontend/simple/components/VotingBanner.vue (hunk 4 above)
Comment (sandrina-p) [reply frontend/simple/components/VotingBanner.vue]: Sure. I'll do it

File: frontend/simple/sass/main.sass (hunk 3 above)
Comment (sandrina-p) [reply frontend/simple/sass/main.sass]: Hum... so should we create our own class instead of using `.is-centered` from bulma?

File: frontend/simple/sass/bulma_on_groupincome/notifications.scss (hunk 5 above)
Comment (sandrina-p) [reply frontend/simple/sass/bulma_on_groupincome/notifications.scss]: Yes, on my personal projects I use [stylelint](https://stylelint.io/), what do you think? I'll fix it it on this file

File: frontend/simple/sass/_groupincome.scss (hunk 9 above)
Comment (sandrina-p) [reply frontend/simple/sass/_groupincome.scss]: For clarity purposes, I believe we should keep separated framework classes from our owns. If we merge them all, it might get confused _"Is this from bulma or not?"_ Maybe the location isn't the best at the moment, but my concern were would be to keep them separated.

File: frontend/simple/sass/main.sass (hunk 3 above)
Comment (hubudibu) [reply frontend/simple/sass/main.sass]: looking further into text modifiers, I think this is what you were looking for: https://versions.bulma.io/0.4.4/documentation/modifiers/helpers/ luckily our current version and the new ones also have the `.has-text-centered` typography helper class for this, so no need for our own

File: frontend/simple/sass/bulma_on_groupincome/notifications.scss (hunk 5 above)
Comment (hubudibu) [reply frontend/simple/sass/bulma_on_groupincome/notifications.scss]: 👍 sounds great! a quick search also said that it could work with .vue files as well 👍👍 should we create a separate issue for that? how tricky is it to set it up?

File: frontend/simple/sass/_groupincome.scss (hunk 9 above)
Comment (hubudibu) [reply frontend/simple/sass/_groupincome.scss]: ok, that totally makes sense 👍 then let's keep it here for now

File: frontend/simple/sass/bulma_on_groupincome/notifications.scss (hunk 5 above)
Comment (sandrina-p) [reply frontend/simple/sass/bulma_on_groupincome/notifications.scss]: Not that tricky. Just search for the right configuration for our needs, install it, add it to our project and fix the current issues. Check #371 😄

File: frontend/simple/components/ProgressOverview.vue [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+</style>
+
+
+<script>
+  export default {
+    name: 'ProgressOverview',
+    data () {
+      return {
+        period: 'July',
+        themeColor: '#319d2f', // The summary and bar color will adjust :D
```
Comment (sandrina-p) [inline frontend/simple/components/ProgressOverview.vue]: @hubudibu @taoeffect   
After playing around with this progressBar and having in mind both "simplicity" and "theme/personalization" concepts, I came with a progress bar that adjusts it's color palette based on the user's theme (using only CSS). 

- "greenish" palette
image

- "purplish" palette
image


I think this approach is better than  the typical "red/bad to green/good". For example, imagine your group is in the first week of the month and the progress is still at 10%. Showing a red bar can be harsh and demotivate the group. It's their first week, so it's normal they are still on 10%. Ofc we could do the logic to calculate the average vs speed of contribution to decide the color, but I think it's more important for the user to have a feel of group personalization than "bad/good" progress.

----

**Update on next day** I've been thinking while developing the dashboard.
@greg you said on our last meeting, you'd like to see some mouse interaction, so the user can understand the meaning of each bar, but I think with the cleaner layout I've suggested (see comment on MR), there's not a priority at this precise moment. Instead, I suggest to finish designing a clean _WYSIWYG_ layout with all needed features and once we have everything balanced we can better decide what interactions should be built and how should they work.
In other words, I think is better we build the dashboard by layers of iterations, instead of focusing too much on one of the points, like a painting. What do you think? cc @aniko

Comment (taoeffect) [reply frontend/simple/components/ProgressOverview.vue]: @sandrina-p This looks fantastic. Approve! 👍

**Comment by taoeffect** [comment]
I'll leave more comments in a bit, but just wanted to give you quick feedback on the latest screenshots: while I was initially thrilled with the direction of the progress-bar colors, now that I see it in context, composed with the rest of the UI, I'm no longer so sure.

Indeed, if you take the progress-bar in isolation, it looks good.

But now having it in the UI, which has purple nowhere else, it looks extremely out of place. But even more so, the `Support History` boxes, with the vertical progress bars, those need to stay as they were.

I totally get it not being so important to have "red = bad", "yellow = mediocre", "green = good" being emphasized in the horizontal progress bar, but in the Support History it is very important.

So, in summary:

- The purple does not work, at all, with our current "kindergarten professional" theme, and so it needs to go.
- Having the horizontal progress-bar be one color is maybe acceptable, **but**, it must be green, and one of the shades (probably the primary shade), needs to be the same shade of green as the green in our logo.
- It might be better to even have the horizontal progress bar change its primary color from all shades of red, to all shades of yellow, to all shades of green, as it fills up, just like the vertical progress-bars.
- The vertical support history progress bars must show different colors for different levels.

In general, I think having the entire horizontal progress bar go through the same range of colors (red, yellow, green) as the horizontal progress bars, makes intuitive sense, as it will connect those two parts of the UI together, and it will remain faithful to the colors we've chosen for "kindergarten professional".

**Comment by taoeffect** [comment]
I'm guessing there was a misunderstanding before. When you showed the single-color horizontal progress bar, I thought you were objecting to having *different* colors in the horizontal progress bar, and yes, I did (and do) agree that it's better to have a single color (and different shades of that color) in the progress-bar.

But for this *first theme*, which is based on the colors in our logo, we want to stick with the design we had, and yes that does mean having *different colors for the progress bars*, not within the progress bar, but as the progress bar fills up (cycling the base color from red, to yellow, to green at <1/3, <2/3, and >2/3).

**Comment by sandrina-p** [comment]
@taoeffect My bad I didn't explain myself well before. My point was exactly that: 
> to having different colors in the horizontal progress bar.

Here the purpose is to have defined just the markup & layout to be flexible enough to support dynamic themes. Whatever color the design has now, it's not the final color, since we still have a lot to readjust our palette colors as mentioned on #373. I completely agree with having default GIS colors theme. We need to work on that as well to make sure those colors work well (good contrast).

Regarding the Suppport History, keeping the bad/good approach is safer in fact.

**Comment by taoeffect** [comment]
So I was not able to run this locally. Here's the error I got:

```
Running "exec:stylelint" (exec) task

frontend/simple/js/ignored/EventLog.vue
 58:9  ✖  Expected single space before "{"   block-opening-brace-space-before

frontend/simple/js/ignored/vue2-slider.vue
 580:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 581:12  ✖  Expected a leading zero                                                                                                                                          number-leading-zero
 584:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 587:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 590:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 596:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 605:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 612:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 619:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 626:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 632:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 638:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 638:1   ✖  Expected selector ".vue-slider-dot" to come before selector ".vue-slider-wrap.vue-slider-disabled .vue-slider-dot"                                               no-descending-specificity
 648:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 648:1   ✖  Expected selector ".vue-slider-horizontal .vue-slider-dot" to come before selector ".vue-slider-wrap.vue-slider-disabled .vue-slider-dot"                        no-descending-specificity
 651:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 651:1   ✖  Expected selector ".vue-slider-vertical .vue-slider-dot" to come before selector ".vue-slider-wrap.vue-slider-disabled .vue-slider-dot"                          no-descending-specificity
 654:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 654:1   ✖  Expected selector ".vue-slider-horizontal-reverse .vue-slider-dot" to come before selector ".vue-slider-wrap.vue-slider-disabled .vue-slider-dot"                no-descending-specificity
 657:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 657:1   ✖  Expected selector ".vue-slider-vertical-reverse .vue-slider-dot" to come before selector ".vue-slider-wrap.vue-slider-disabled .vue-slider-dot"                  no-descending-specificity
 660:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 674:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 674:44  ✖  Expected single space before "{"                                                                                                                                 block-opening-brace-space-before
 679:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 679:47  ✖  Expected single space before "{"                                                                                                                                 block-opening-brace-space-before
 684:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 684:45  ✖  Expected single space before "{"                                                                                                                                 block-opening-brace-space-before
 689:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 689:46  ✖  Expected single space before "{"                                                                                                                                 block-opening-brace-space-before
 694:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 706:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 718:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 718:53  ✖  Expected single space before "{"                                                                                                                                 block-opening-brace-space-before
 730:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 730:54  ✖  Expected single space before "{"                                                                                                                                 block-opening-brace-space-before
 742:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 745:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 745:1   ✖  Expected selector ".vue-slider-dot.vue-slider-always .vue-slider-tooltip" to come before selector ".vue-slider-dot.vue-slider-hover:hover .vue-slider-tooltip"   no-descending-specificity
 746:17  ✖  Expected single space before "!"                                                                                                                                 declaration-bang-space-before
 748:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 758:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 763:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 773:5   ✖  Expected indentation of 2 spaces                                                                                                                                 indentation
 774:19  ✖  Expected a leading zero                                                                                                                                          number-leading-zero
 776:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before
 776:63  ✖  Expected newline after ","                                                                                                                                       selector-list-comma-newline-after
 779:1   ✖  Expected empty line before rule                                                                                                                                  rule-empty-line-before

>> Exited with code: 2.
>> Error executing child process: Error: Process exited with code 2.
Warning: Task "exec:stylelint" failed. Use --force to continue.
```

In the meantime I'm going to go work on some tooling issues for now while this is figured out (Docker: #375, and ability to run two GIS instances: #378)

**Comment by taoeffect** [comment]
@sandrina-p FYI I get the same errors locally. Can you configure `exec:stylelint` to ignore the files in `frontend/simple/js/ignored` somehow?

**Comment by taoeffect** [comment]
So I think I fixed the stylelint errors in PR #379 ^_^

**Comment by sandrina-p** [comment]
@hubudibu and @taoeffect I've just updated with master so you can finish your review 😄

File: frontend/simple/components/GroupMembers.vue [hunk 11]
```diff
@@ -1,37 +1,109 @@
...
+            <span class="icon"><i class="fa fa-plus has-text-success"></i></span>
+          </a>
+        </div>
+      </div>
+    </div>
   </section>
 </template>
 <style lang="scss" scoped>
-  h3.title {
+  // @import "bulma/sass/utilities/mixins";
```
Comment (hubudibu) [inline frontend/simple/components/GroupMembers.vue:36]: you could either remove this comment or add some more info, a TODO on why we want to keep it around. same goes for the commented out include

File: frontend/simple/components/ProgressOverview.vue [hunk 12]
```diff
@@ -0,0 +1,129 @@
+<template>
+  <section>
+    <div class="level">
+      <h3 class="title"><i18n>{{ period }} Overview</i18n></h3>
+    </div>
+      <div class="level">
+        <dl class="level-left summary" :style="{ color: themeColor }">
```
Comment (hubudibu) [inline frontend/simple/components/ProgressOverview.vue:7]: I guess this will also rather be handled by a theme class rather than a style attribute as in #373 right? or should theming be 100% flexible?

File: frontend/simple/components/SupportHistory.vue [hunk 13]
```diff
@@ -0,0 +1,107 @@
+<template>
+  <section>
+    <h3 class="title"><i18n>Support History</i18n></h3>
+    <p v-if="history.length === 0">Your group is still in its first month.</p>
+    <div class="history columns" v-else>
+      <div v-for="(percentage, index) in history" class="column is-2">
+        <div :class="['period', getResult(percentage)]">
+          <p class="period-title">{{ months[index] }}</p>
+          <p class="period-txt">{{ Math.floor(percentage * 100) }}%</p>
```
Comment (hubudibu) [inline frontend/simple/components/SupportHistory.vue:9]: we will have a nice filter to format percentage values once the #355 sbp pr is accepted, leaving this here so I don't forget to apply that here once this is merged :)

**Review by hubudibu — APPROVED** [review]
👍 👍 👍 🎉 
just a few questions, but it's good to go anyways!

File: frontend/simple/components/GroupMembers.vue (hunk 11 above)
Comment (sandrina-p) [reply frontend/simple/components/GroupMembers.vue:36]: In fact I've forgot to submit a comment regarding your comment about my previous commented code

```js
// How to I import missins from Bulma to be available here?
// Bulma mixins are in SASS but this is SCSS... ._.
```
> @hubudibu hubudibu 11 days ago  Member
> the syntax difference in itself shouldn't be a problem I think.
> we could probably import the bulma mixins file in the components where it's used

Well... I've tried to import those mixins but it didn't work, that's why I've leaved it commented, hoping you can clarify me on that.

File: frontend/simple/components/ProgressOverview.vue (hunk 12 above)
Comment (sandrina-p) [reply frontend/simple/components/ProgressOverview.vue:7]: You are right, I leaved it as it is to better rethink it on a next iteration.

File: frontend/simple/components/GroupMembers.vue (hunk 11 above)
Comment (sandrina-p) [reply frontend/simple/components/GroupMembers.vue:36]: Nevermind, I've found it, it was just `@import "bulma";` Bulma at this current version don't support modular imports yet, so on #377 I'll fix that up to use only the CSS we need, fixing part of #101 issue
